docs/source/common_issues.rst

```diff
@@ -821,3 +821,32 @@ This is best understood via an example:
 To get this code to type check, you could assign ``y = x`` after ``x`` has been
 narrowed, and use ``y`` in the inner function, or add an assert in the inner
 function.
+
+
+Upgrading to PEP584 syntax
+---------------------------
+
+In pre-PEP584 syntax type annotations for builtin types such as ``dict``, ``list``, ``tuple``, ``type`` were taken from the ``typing`` module as ``typing.Dict``, ``typing.List``, ``typing.Tuple``, ``typing.Type`` etc.
+
+When upgrading a codebase you may hit a case such as the following:
+
+.. code-block:: python
+
+    from typing import TypeVar
+
+    T = TypeVar('T', bound='Foo')
+
+    class Foo:
+	type: str
+
+	@classmethod
+	def foo(cls: type[T], value: str) -> None:  # will fail due to "not a valid type"
+	    pass
+
+This is happening since the class has an attribute that is overshadowing the built-in keyword ``type``. To get this code to type-check you have the following options:
+
+- Use ``typing.Type`` and use ``Type[]`` for your annotation.
+- Stick ``import builtins`` at the top and use ``builtins.type[]`` for your annotation
```

Comment:
My wording in the issue was slightly informal; that's fine for an issue, but we can do better in docs :)

```suggestion
- Add ``import builtins`` to the top of your module and use ``builtins.type[]`` for your annotation
```